Userland/Libraries/LibGfx/Painter.h

```diff
@@ -176,19 +176,15 @@ class Painter {
     template<typename TGetPixelCallback>
     void fill_pixels(Gfx::IntRect const& region, TGetPixelCallback callback, bool blend = false)
     {
-        // Note: This function paints physical pixels and therefore does not make sense when
-        // called on a scaled painter. Scaling the region painted may break the caller (this
-        // would be the case for gradients in LibWeb), and if you scale the pixels (to squares)
-        // then this is no longer filling pixels.
-        VERIFY(scale() == 1);
-        auto paint_region = region.translated(translation());
-        auto clipped_region = paint_region.intersected(clip_rect());
+        auto scaled_region = region * scale();
+        auto paint_region = region.translated(translation()) * scale();
+        auto clipped_region = paint_region.intersected(clip_rect()) * scale();
         if (clipped_region.is_empty())
             return;
         auto start_offset = clipped_region.location() - paint_region.location();
         for (int y = 0; y < clipped_region.height(); y++) {
             for (int x = 0; x < clipped_region.width(); x++) {
-                auto pixel = callback(IntPoint(x, y).translated(start_offset));
+                auto pixel = callback(FloatPoint(x, y).translated(start_offset.to_type<float>()).divided_individually_by(scaled_region.width(), scaled_region.height()));
```

Comment:
I can't say I'm a super fan of this as it's adding a division back into the hot loop of CSS gradients, and also possible floating point precision stuff since that only cares about integer pixels (you can't really scale the painter for CSS gradients as it only generate points for the pixels they know about). 

I did consider something like this, but I'd much prefer it if it used template magic to infer if the callback takes integers and thus only scales about pixels, or if it's something that can scale.

Reply:
And I just noticed that the precision is already causing problems.. (see the radial gradients)

Reply:
I would like to get rid of/don't even introduce scaling outside LibGfx.
That's the main motivation to do it this way.
This could be refactored to only iterate over x with `x = x + 1/scaled_region.width()`, now that would bring its own rounding problems, but should be more performant.

Reply:
I removed the division in the hot loop.